kod/object/active/holder/nomoveon/battler/player/user.kod

```diff
@@ -3627,26 +3637,38 @@ messages:
               
          return;
       }
-        
-      if NOT Send(self,@ReqNewHold,#what=what) 
+      % Either we pass in a number or we didn't but can't hold it all
+      if number <> $
+         OR NOT Send(self,@ReqNewHold,#what=what)
```

Comment:
Is there a security problem here?  I'm looking for where we check that the number received from the client is at most the number that actually exist.  It might fail at the call to Split with a bunch of errors, but it would be much better to have an explicit check.

Reply:
So I did test something along these lines: the game doesn't let you enter a number greater than the available quantity.  I am not sure where that code lives and will find it and see how robust it is.  

It may be possible that a bad actor could hack the client to circumvent this.  I will look into the Vaultman code and see if I can find some method for checking that the number of items exist as a secondary check.

Reply:
Added an explicit check for if Split() returns $.  This also checks if somehow the user enters a number larger than the number of items available in the container.